language/tools/move-package/src/compilation/compiled_package.rs

```diff
@@ -463,142 +464,72 @@ impl CompiledPackage {
+            .collect();
+        let root_package_name = resolved_package.source_package.package.name;
+        writeln!(w, "{} {}", "BUILDING".bold().green(), root_package_name)?;
 
         // gather source/dep files with their address mappings
         let (sources_package_paths, deps_package_paths) = make_source_and_deps_for_compiler(
             resolution_graph,
             &resolved_package,
-            deps_source_info,
+            transitive_dependencies,
         )?;
         let flags = if resolution_graph.build_options.test_mode {
             Flags::testing()
         } else {
             Flags::empty()
         };
         // invoke the compiler
-        let compiler = Compiler::from_package_paths(
-            vec![sources_package_paths.clone()],
-            deps_package_paths.clone(),
-        )
-        .set_flags(flags);
-        let (file_map, compiled_units) = compiler_driver(compiler, is_root_package)?;
-
-        let (compiled_units, resolutions): (Vec<_>, Vec<_>) = compiled_units
-            .into_iter()
-            .map(|annot_unit| {
-                let originating_file =
-                    PathBuf::from(file_map[&annot_unit.loc().file_hash()].0.as_str());
-                match &annot_unit {
-                    AnnotatedCompiledUnit::Module(annot_module) => {
-                        // Only return resolutions for modules that have named addresses
-                        let resolution = match annot_module.module_id() {
-                            (Some(str_name), module_id) => Some((module_id, str_name.value)),
-                            _ => None,
-                        };
-                        (
-                            CompiledUnitWithSource {
-                                unit: annot_unit.into_compiled_unit(),
-                                source_path: originating_file,
-                            },
-                            resolution,
-                        )
-                    }
-                    AnnotatedCompiledUnit::Script(_) => (
-                        CompiledUnitWithSource {
-                            unit: annot_unit.into_compiled_unit(),
-                            source_path: originating_file,
-                        },
-                        None,
-                    ),
-                }
-            })
-            .unzip();
-
-        // gather module resolution data from deps
-        let mut module_resolution_metadata = BTreeMap::new();
-        for dep_package in &dependencies {
-            dep_package.gather_module_resolution_metadata(
-                &resolved_package.renaming,
-                &mut module_resolution_metadata,
-            )
-        }
-        // gather from compiled units (which do not include deps)
-        for (mod_id, name) in resolutions.into_iter().flatten() {
-            let _old_value = module_resolution_metadata.insert(mod_id, name);
-            debug_assert!(_old_value.is_none());
+        let paths = {
```

Comment:
Am I reading it right that we will now have all ASTs for all (dependent) packages generated with a single compiler invocation without any additional modifications?

Reply:
Correct. And each definition (module or script) will have the package name tagged on top of it, in the field `package_name: Option<Symbol>`